htdocs/compta/facture.php

```diff
@@ -3533,6 +3527,14 @@
 		}
 	}
 
+	// Add products form
+	print '	<form name="addproduct" id="addproduct" action="' . $_SERVER["PHP_SELF"] . '?id=' . $object->id . (($action != 'editline') ? '#add' : '#line_' . GETPOST('lineid')) . '" method="POST">
```

Comment:
You move a form tag inside a table. form open tag must be before the table open tag.
You you need a form for situation invoice, you must use same form and add a name on button to differentiate action when submiting.